src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +25,29 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        File file = new File(fileName);
+        if (!file.exists()) {
+            throw new FileNotFoundException("File is not found");
+        }
+        StringBuilder fileData = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                fileData.append(line.toLowerCase());
+            }
+        }
+        Set<Character> tree = new TreeSet<>();
+        for (int i = 0; i < fileData.length(); i++) {
+            Character symbol = fileData.charAt(i);
+            if (Character.isAlphabetic(symbol)) {
+                tree.add(symbol);
+            }
+        }
+        StringBuilder result = new StringBuilder();
+        for (Character letter : tree) {
+            result.append(letter);
+        }
+        return result.length() < 6 ? result.toString() : result.toString().substring(0, 5);
```

Comment:
3 loops that's too much) And think if we need to add all characters to stringbuilder ?

Reply:
done